File: server/Models/Keep.cs
namespace keeper.Models;

public class Keep : RepoItem<int>
{
    public string Name { get; set; }
    public string Description { get; set; }
    public string Img { get; set; }
    public int Views { get; set; }
    public int Kept { get; set; }
    public string CreatorId { get; set; }
    public Account Creator { get; set; }

}
public class VaultKept : Keep
{
    public int VaultKeepId { get; set; }
}

File: client/src/models/Keep.js
export class Keep {
    constructor(data) {
        this.id = data.id
        this.name = data.name
        this.description = data.description
        this.img = data.img
        this.views = data.views
        this.kept = data.kept
        this.creatorId = data.creatorId
        this.creator = data.creator
    }
}

export class VaultKept extends Keep {
    constructor(data) {
        super(data)
        this.vaultKeepId = data.vaultKeepId
    }
}